src/tournament/roles.rs

```diff
@@ -0,0 +1,165 @@
+use std::{fmt, str::FromStr};
+
+use serde::{Deserialize, Serialize};
+use sqlx::{query, Pool, Postgres};
+use strum::VariantArray;
+use tracing::info;
+use utoipa::ToSchema;
+use uuid::Uuid;
+
+use crate::{omni_error::OmniError, users::permissions::Permission};
+
+#[derive(Debug, PartialEq, Deserialize, ToSchema, VariantArray, Clone, Serialize)]
+#[serde(deny_unknown_fields)]
+/// Within a tournament, users must be granted roles for their
+/// permissions to be defined. Each role comes with a predefined
+/// set of permissions to perform certain operations.
+/// By default, a newly created user has no roles.
+/// Multiple users can have the same role.
+pub enum Role {
+    /// This role grants all possible permissions within a tournament.
+    Organizer,
+    /// Judges can submit their verdicts regarding debates they were assigned to.
+    Judge,
+    /// Marshalls are responsible for conducting debates.
+    /// For pragmatic reasons, they can submit verdicts on Judges' behalf.
+    Marshall,
+}
+
+impl Role {
+    pub fn get_role_permissions(&self) -> Vec<Permission> {
+        use Permission as P;
+        match self {
+            Role::Organizer => P::VARIANTS.to_vec(),
+            Role::Judge => vec![
+                P::ReadAttendees,
+                P::ReadDebates,
+                P::ReadTeams,
+                P::ReadTournament,
+                P::SubmitOwnVerdictVote,
+                P::ReadLocations,
+                P::ReadRooms,
+            ],
+            Role::Marshall => vec![
+                P::ReadDebates,
+                P::ReadAttendees,
+                P::ReadTeams,
+                P::ReadTournament,
+                P::ReadLocations,
+                P::ReadRooms,
+                P::SubmitVerdict,
+            ],
+        }
+    }
+
+    pub async fn post(
+        user_id: Uuid,
+        tournament_id: Uuid,
+        roles: Vec<Role>,
+        pool: &Pool<Postgres>,
+    ) -> Result<Vec<Role>, OmniError> {
+        let _ = tournament_id;
+        let roles_as_strings = Role::roles_vec_to_string_vec(&roles);
```

Comment:
Would be nicer to implement a `Vec<Role>` to `Vec<String>` converter via an extending trait, rather than make a method on Role that takes in a Vec as arg...
See suggestion and relevant impl below.
```suggestion
        let roles_as_strings = roles.to_string_vec();
```